### PR TITLE
Replace GetElementOrDefault with GetValidElement

### PR DESCRIPTION
**Pull request by nnikolopoulos**
Instead of:

`GetElementOrDefault<T>(this ICollection<T> array, int index)`

have:

`GetValidElement<T>(this IList<T> array, int index)`.

### CODE REVIEW

File: ifrs17/Utils/Extensions.ipynb [hunk 1]
```diff
@@ -75,8 +75,7 @@
         {
             "cell_type": "code",
             "source": [
-                "// because the default(T) is something else than the first/last element. What about \"static T GetValidElement<T>(this IList<T> array, int index)\"? ",
-                "\nstatic T GetElementOrDefault<T>(this ICollection<T> array, int index)",
+                "static T GetValidElement<T>(this IList<T> array, int index)",
```
Comment (dcolleoni) [inline ifrs17/Utils/Extensions.ipynb:78]: why the change in type? from ICollection to IList?

Comment (nnikolopoulos) [reply ifrs17/Utils/Extensions.ipynb]: I reverted this change

File: ifrs17/Import/ImportScopeCalculation.ipynb [hunk 2]
```diff
@@ -805,11 +805,11 @@
                 "\n        switch (periodType) {",
                 "\n            case PeriodType.BeginningOfPeriod :",
                 "\n                for (var i = 0; i < parentDiscountedValues.Length; i++)",
-                "\n                     ret[i] = -1d * (parentDiscountedValues[i] - parentNominalValues[i]) * (GetElementOrDefault(monthlyInterestFactor, i/12) - 1d );",
+                "\n                     ret[i] = -1d * (parentDiscountedValues[i] - parentNominalValues[i]) * (GetValidElement(monthlyInterestFactor, i/12) - 1d );",
```
Comment (dcolleoni) [inline ifrs17/Import/ImportScopeCalculation.ipynb:808]: let's make use of "this"
=> monthlyInterestFactor.GetValidElement(i/12)

Comment (nnikolopoulos) [reply ifrs17/Import/ImportScopeCalculation.ipynb]: Good idea, I applied the proposed syntax in every call of `GetValidElement`.

File: ifrs17/Constants/Validations.ipynb [hunk 3]
```diff
@@ -169,6 +171,8 @@
                 "\n    (Error.MissingDataAtPosting           , 1) => $\"Missing imported data for {s[0]} DataNode.\",",
                 "\n    (Error.MissingCombinedLiability       , 2) => $\"Missing Combined Liability AoC Type for DataNode {s[0]} and AmountType {s[1]}.\",",
                 "\n    (Error.MissingCoverageUnit            , 1) => $\"Missing Coverage Unit cash flow for {s[0]} DataNode.\",",
+                "\n    // Index",
+                "\n    (Error.NegativeIndex                  , 0) => $\" Index was out of range. Must be non-negative. (Parameter 'index')\",",
```
Comment (dcolleoni) [inline ifrs17/Constants/Validations.ipynb:175]: You can remove  (Parameter 'index')

**Review by dcolleoni — APPROVED** [review]
